0000-stream/0000-stream.md

```diff
@@ -0,0 +1,407 @@
+
+Each endpoint can limit the amount of money and data they are willing to receive on a particular stream. STREAM uses a credit-based flow control scheme inspired by [QUIC](https://quicwg.github.io/base-drafts/draft-ietf-quic-transport.html#rfc.section.10). Each endpoint advertises the maximum amount of money they are willing to receive, using `StreamMaxMoney` frames, as well as the maximum number of bytes each stream can receive, using `StreamMaxData` frames.
+
+Endpoints MAY advertise larger offsets at any point by sending new `StreamMaxMoney` or `StreamMaxData` frames. An endpoint MUST NOT renege on an advertisement. Once an endpoint advertises a given maximum receive amount or maximum byte offset, they MUST NOT advertise a smaller value later. The sending endpoint could receive the frames out of order and so they must ignore flow control offsets that do not increase the window.
+
+The receiving endpoint MUST close the connection with a `FlowControlError` if the sender violates the advertised stream limits.
+
+Senders SHOULD send `StreamMoneyBlocked` and `StreamDataBlocked` frames when they have more money or data to send that would exceed the other endpoint's advertised limits. These are primarily intended for debugging purposes.
+
+#### 4.4.5. Closing Streams
+
+Either endpoint can close a stream using a `StreamClose` frame. Implementations MAY allow half-open streams (where one side has closed and the other is still able to send).
+
+`StreamClose` frames are used to communicate both normal stream closes as well as errors.
+
+### 4.5. Connection-Level Flow Control
+
+Similar to the [stream-level flow control](#444-stream-level-flow-control), endpoints can limit the total amount of incoming data on all streams. Endpoints advertise the total number of bytes they are willing to receive on a given connection using `ConnectionMaxData` frames. Endpoints MAY increase the advertised limits by sending additional `ConnectionMaxData` frames with higher total byte limits.
+
+The receiving endpoint MUST close the connection with a `FlowControlError` if the sender violates the advertised limit.
+
+### 4.6. Closing Connections
+
+Either endpoint can close the connection using a `ConnectionClose` frame. Implementations MAY allow half-open connections (where one side has closed the connection and the other is still able to send).
+
+`ConnectionClose` frames are used to communicate both normal connection closes as well as errors.
+
+## 5. Packet and Frame Specification
+
+STREAM packets are encrypted and attached to ILP Prepare, Fulfill, and Reject packets as the `data` field.
+
+STREAM uses the [Octet Encoding Rules (OER)](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview_of_OER.pdf) to encode packet fields.
+
+### 5.1. Encryption
+
+All STREAM packets are encrypted using [AES-256-GCM](https://en.wikipedia.org/wiki/Galois/Counter_Mode) with a 12-byte Initialization Vector (IV) and a 16-Byte Authentication Tag.
```

Comment:
Should this describe how to encode these into the ILP address?

Reply:
That isn't in the ILP address, those values are in the encryption envelope in the ILP packet data